File: course-material/labs/extra/movie/solution/date.cpp
#include "date.h"


Date::Date(int day, int month, int year)
    : day(day), month(month), year(year) { }


bool Date::operator==(const Date& date) const
{
    return day == date.day && month == date.month && year == date.year;
}

bool Date::operator!=(const Date& date) const
{
    return !(*this == date);
}

std::ostream& operator <<(std::ostream& out, const Date& date)
{
    return out << date.day << "/" << date.month << "/" << date.year;
}

File: course-material/labs/extra/movie/java/Date.java
public class Date
{
    public int day, month, year;

    public Date(int day, int month, int year)
    {
        this.day = day;
        this.month = month;
        this.year = year;
    }

    // Implement as operator == and operator !=
    public boolean equals(Date that)
    {
        return this.day == that.day && this.month == that.month && this.year == that.year;
    }

    // Implement as << on std::ostream
    public String toString()
    {
        return String.format("%d/%d/%d", day, month, year);
    }       
}
